nextmove/src/main/java/no/difi/meldingsutveksling/nextmove/DpfConversationResource.java

```diff
@@ -1,15 +1,51 @@
 package no.difi.meldingsutveksling.nextmove;
 
 import lombok.Data;
+import lombok.NoArgsConstructor;
 
 import javax.persistence.DiscriminatorValue;
 import javax.persistence.Entity;
+import java.time.LocalDateTime;
+import java.util.Map;
+
+import static no.difi.meldingsutveksling.ServiceIdentifier.DPF;
 
 @Entity
 @DiscriminatorValue("DPF")
+@NoArgsConstructor
 @Data
 public class DpfConversationResource extends ConversationResource {
 
-    DpfConversationResource() {}
+    private DpfConversationResource(String conversationId, String senderId, String receiverId) {
+        super(conversationId, senderId, receiverId, DPF);
+    }
+
+    private DpfConversationResource(String conversationId,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00107)